src/server/worker/pipeline/transform/pending_job.go

```diff
@@ -160,12 +159,30 @@ func (pj *pendingJob) withDeleter(pachClient *client.APIClient, cb func(datum.De
 	})
 }
 
+// writeDatumCount gets the total count of input datums for the job and updates
+// its DataTotal field.
+func (pj *pendingJob) writeDatumCount(ctx context.Context, taskDoer task.Doer) error {
+	pachClient := pj.driver.PachClient().WithCtx(ctx)
+	return pachClient.WithRenewer(func(ctx context.Context, renewer *renew.StringSet) error {
```

Comment:
nit, the scope of the renewer could be reduced to just the create full job datum file set.